go/tasks/plugins/k8s/sidecar/sidecar.go

```diff
@@ -62,6 +62,7 @@ func validateAndFinalizeContainers(
 	}
 	pod.Spec.Containers = finalizedContainers
 	pod.Spec.Tolerations = flytek8s.GetPodTolerations(taskCtx.TaskExecutionMetadata().IsInterruptible(), resReqs...)
+	pod.Spec.SchedulerName = flytek8s.GetPodTolerations(taskCtx.TaskExecutionMetadata().SchedulerName(), resReqs...)
```

Comment:
@EngHabu @surindersinghp @migueltol22 is it correct to update sidecar with this option?

Should I also update go/tasks/plugins/k8s/spark/spark.go?

Reply:
This shouldn't be necessary. If my understanding is correct this value will already be populated for all tasks that run on kubernetes if set, so we shouldn't need to modify any of the tasks.